src/PathMappingRepository.php

```diff
@@ -107,7 +110,11 @@ public function listChildren($path)
      */
     public function hasChildren($path)
     {
-        return $this->getChildren($this->get($path), false)->count() !== 0;
+        if (!$this->resolveResource($path)) {
```

Comment:
IMO you should implement another method `isResourceResolvable()` which returns `true` or `false` for performance reasons. You are doing this check quite often.